### PR TITLE
Issue zaproxy/zaproxy#1790  Move Buffer Overflow Scanner from Beta to Release

### PR DESCRIPTION
**Pull request by msrader**
Attached is the buffer overflow code for change from Beta to release.

### CODE REVIEW

**Comment by thc202** [comment]
Could you rebase the PR?

**Comment by msrader** [comment]
It shows 1 commit with 4 files and up to date with the base branch

**Comment by kingthorin** [comment]
https://wiki.eclipse.org/EGit/User_Guide#Interactive_Rebase

**Comment by msrader** [comment]
That doesn't work in this case.  After you have done a merge, it seems like you can squash anything that happens before the merge, or after the merge.

**Comment by thc202** [comment]
Yes, it seems EGit interactive rebase preserves the merge commits.
You can rebase interactive with ```git``` though, which by default does not keep the merge commits.
The ```changes``` of  ```ZapAddOn.xml``` were lost during the merge conflict.

**Comment by msrader** [comment]
The changes have been re-added.  I'm trying to figure out how to do the rebase with the command line.

**Comment by msrader** [comment]
At a certain point just re-adding the changes is easier.

**Comment by thc202** [comment]
It should also be easier to rebase ;)
Give me some minutes to try with your repo.

**Comment by thc202** [comment]
OK. Following steps to just have your changes. Would you mind giving it a try?
(You should open the PR though)
To start rebase interactive (in the root directory of the repo and with master checked out):
```git rebase -i d3951871790d00451efb8b0f13fa232b9aadc5ac``` (this is the commit where we want to rebase)
It should open a text editor where it should have just your latest commit.
Close the editor and it should start the rebase.
It should stop with a conflict, edit the ```ZapAddOn.xml``` file so that it has the changes that you want.
Add it to the index:
```git add src/org/zaproxy/zap/extension/ascanrules/ZapAddOn.xml```
Continue the rebase:
```git rebase --continue```
It should open the editor again so you can change the commit message, close it (there's nothing to add).
It should finish the rebase and the branch should have just one commit with just your files, ready to push.

**Comment by thc202** [comment]
If your local repo does not have the commit ```d3951871790d00451efb8b0f13fa232b9aadc5ac``` you need to fetch from upstream first.

**Comment by thc202** [comment]
btw, you might need to reset the author of the commit, it seems that it picked up another author in a previous rebase.
Before pushing, run:
```git commit --amend --reset-author```
close the editor and ready for push.

**Comment by msrader** [comment]
Ok, I think it worked.  Thanks for the help.

File: src/org/zaproxy/zap/extension/ascanrules/ZapAddOn.xml [hunk 1]
```diff
@@ -6,12 +6,16 @@
 	<url/>
 	<changes>
 	<![CDATA[
+        Improved "Path Traversal" scanner.<br>
+        Added Buffer Overflow scanner. <br>
+        Change scanners to honour the technologies enabled.
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/ascanrules/ZapAddOn.xml]: It should have just the "Buffer Overflow" entry.

**Comment by thc202** [comment]
No problem! Glad it worked. There's just a minor change left.

**Comment by msrader** [comment]
Done.

File: src/org/zaproxy/zap/extension/ascanrules/ZapAddOn.xml [hunk 2]
```diff
@@ -1,35 +1,37 @@
-<zapaddon>
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/ascanrules/ZapAddOn.xml]: It seems that line endings were changed from ```\n``` to ```\r\n```.
Could you replace the line endings with the former?

**Comment by msrader** [comment]
Ahh the fun of windows.

**Comment by msrader** [comment]
Done.  Eclipse actually has a function for that.  I did a linefeed conversion to UNIX.  Can't see it though with the editor I have.

File: src/org/zaproxy/zap/extension/ascanrules/ZapAddOn.xml [hunk 3]
```diff
@@ -1,17 +1,19 @@
 <zapaddon>
 	<name>Active scanner rules</name>
-	<version>20</version>
+	<version>18</version>
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/ascanrules/ZapAddOn.xml]: The version was changed.

**Comment by thc202** [comment]
Line endings are correct.

**Comment by msrader** [comment]
Version number corrected.

**Comment by thc202** [comment]
Thank you!

**Comment by msrader** [comment]
You are welcome

**Pull request merged**